src/pages/market/components/AllItemsSection.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react';
+import { getProducts } from '../../../api/itemApi';
+import ItemCard from './ItemCard';
+import { ReactComponent as SortIcon } from '../../../images/icons/ic_sort.svg';
+import { ReactComponent as SearchIcon } from '../../../images/icons/ic_search.svg';
+import DropdownList from '../ui/DropdownList';
+import PaginationBar from '../ui/PaginationBar';
+
+const getPageSize = () => {
+  const width = window.innerWidth;
+  if (width < 768) {
+    // Mobile viewport
+    return 4;
+  } else if (width < 1280) {
+    // Tablet viewport
+    return 6;
+  } else {
+    // Desktop viewport
+    return 10;
+  }
+};
+
+function AllItemsSection() {
+  const [orderBy, setOrderBy] = useState('recent');
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(getPageSize());
+  const [itemList, setItemList] = useState([]);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [totalPageNum, setTotalPageNum] = useState();
+
+  const fetchSortedData = async ({ orderBy, page, pageSize }) => {
+    const products = await getProducts({ orderBy, page, pageSize });
+    setItemList(products.list);
+    setTotalPageNum(Math.ceil(products.totalCount / pageSize));
+  };
+
+  const handleSortSelection = (sortOption) => {
+    setOrderBy(sortOption);
+    setIsDropdownVisible(false);
+  };
+
+  useEffect(() => {
+    const handleResize = () => {
+      setPageSize(getPageSize());
+    };
+
+    window.addEventListener('resize', handleResize);
+    fetchSortedData({ orderBy, page, pageSize });
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [orderBy, page, pageSize]);
+
+  const toggleDropdown = () => {
+    setIsDropdownVisible(!isDropdownVisible);
+  };
+
+  const onPageChange = (pageNumber) => {
+    setPage(pageNumber);
+  };
+
+  return (
+    <div>
+      <div className='allItemsSectionHeader'>
+        <h1 className='sectionTitle'>판매 중인 상품</h1>
+        <div className='searchBarWrapper'>
+          <SearchIcon />
+          <input className='searchBarInput' placeholder='검색할 상품을 입력해 주세요' />
+        </div>
+        <a href='/additem' className='login button'>
+          상품 등록하기
+        </a>
+        <div className='sortButtonWrapper'>
+          <button className='sortDropdownTriggerButton' onClick={toggleDropdown}>
```

Comment:
className이나 변수 이름짓는건 항상 어렵죠.. 어느선까지 구체적으로 해야할지 애매모호해서 더 자세하게 붙이다보면 본의아니게 길어지곤 합니다.
저는 이름이 길어지면 일단 의미는 지킬 수 있으면서 줄일 수 있는 단어가 있는지 찾는편이에요.

예를들면 
```suggestion
        <div className='sortWrapper'>
          <button className='sortButton' onClick={toggleDropdown}>
```
이렇게요 :-)